### ISSUE TITLE
Consider name generation being configurable.

### ISSUE THREAD

**Issue by bbuck** [layeh/gopher-luar#23]
I love this library, but currently am using a personally modified version of it so that when I inject Go values with methods/fields I get only one style of methods in return. It would be awesome to have the the exported style and un-exported style behind a set of options. The default can be it's current behavior.

Perhaps something along the lines of:

```go
luar.Configure(luar.Options{
        MethodNames: luar.ExportedOnly,
})
```

Other values could be `luar.UnexportedOnly`, or `luar.Default` (I'm drawing a blank at a better name for this one).

**Comment by bbuck** [comment]
@bontibon I see you already assigned this to you, but I went ahead and put what I was thinking into code and made a PR. Would love a code review when you have time. Thanks!

**Comment by bontibon** [comment]
Thanks for your thoughts, @bbuck.

I agree that the enforced dual naming for (at least) methods is not great. Struct fields I'm not as concerned about, as they can be explicitly set with the "luar" tag.

Your proposed solution and PR with the `luar.Configure` function cannot be added, as it modifies global state. The solution needs to operate on types in a specific `*lua.LState`.

A possibility I see is adding a helper to `luar.MT` that removes (or prevents) a specific set of methods.

**Comment by bbuck** [comment]
Where would a good place be to store this information on an `LState`? I've toyed with a similar feature in my current project for passing around values tied directly to an `LState` but I'm not sure how best to store meta information. I started using globals like `_SOME_GLOBAL_NAME` on the state but then, for my use case, chose to wrap stats with a data container so that values can't be modified from within scripts run. 

Long story short, I'm not super familiar with the internals of Lua (yet) but I'm trying to learn. Where do you think might be the best place to store information like this? I'm totally willing to make the change in my PR.

**Comment by bontibon** [comment]
You would want to store such information in the Lua state registry. See `getMTCache` in `cache.go` for an example of how we currently use the registry.

**Comment by bbuck** [comment]
Thanks for doing this, much more compact than the new concept that I had.
